packages/next/next-server/lib/router/router.ts

```diff
@@ -491,18 +493,30 @@ export default class Router implements BaseRouter {
     if (!(options as any)._h && this.onlyAHashChange(cleanedAs)) {
       this.asPath = cleanedAs
       Router.events.emit('hashChangeStart', as)
+      // TODO: do we need the resolved href when only a hash change?
       this.changeState(method, url, as, options)
       this.scrollToHash(cleanedAs)
       this.notify(this.components[this.route])
       Router.events.emit('hashChangeComplete', as)
       return true
     }
 
-    const parsed = tryParseRelativeUrl(url)
+    const pages = await this.pageLoader.getPageList()
```

Comment:
Does this mean auto static generated dynamic pages don't get their query parameters until the build manifest downloads? Can we add a comment indicating that?

Reply:
Right, can add a comment for this